blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+---
+title: Implementing UEFI Secure Boot on MPL PIP4x
+abstract: "This post explains how we tackled the problem of implementing UEFI
+           Secure Boot on MPL's PIP platforms. The process included testing the
+           platform's compatibility with Secure Boot and integrating automatic
+           image signing into an existing Yocto layer."
+cover: /covers/mpl-pip44.jpg
+author: pawel.langowski
+layout: post
+published: true
+date: 2024-03-15
+archives: "2024"
+
+tags:
+  - secure boot
+  - uefi
+  - mpl
+  - yocto
+categories:
+  - Firmware
+  - Security
+
+---
+
+MPL is a Swiss company, which designs and manufactures embedded computers and
+microcontroller hardware for rugged environment, extended temperature range,
+and with long-term availability. The PIP series is a family of low-power,
+read-to-use embedded computers manufactured by MPL. Recently we tackled the
+problem of implementing UEFI Secure Boot on platforms from the PIP4x series. The
+goal of the project was to verify that the platform in question is compatible
+with Secure Boot and to enable automatic signing of system components during
+build in Yocto.
+
+## Verifying Secure Boot compatibility
+
+The first step we took was verifying that Secure Boot could indeed be
+implemented on the platform and identifying potential issues and vulnerabilities.
+To do that, we developed and executed various automated tests within the Dasharo
+OSFV
+([Open Source Firmware Validation](https://github.com/Dasharo/open-source-firmware-validation))
+environment. It utilizes the [Robot Framework](https://robotframework.org/) – an
+open-source automation framework, which simplifies the creation and execution
+of test cases.
+
+### Setup
+
+In order to prepare the host machine for Dasharo OSFV infrastructure, follow
```

Comment:
@PLangowski why not link to osfv README page? https://github.com/Dasharo/open-source-firmware-validation?tab=readme-ov-file#getting-started

Reply:
16f72d0ee96a1a1d242206f9583ebf32126162f6